pkg/cloud/services/compute/instance.go

```diff
@@ -152,19 +152,24 @@ func (s *Service) constructNetworks(openStackCluster *infrav1.OpenStackCluster,
 		}
 	}
 	for i, port := range openStackMachine.Spec.Ports {
+		pOpts := &openStackMachine.Spec.Ports[i]
+		// No Trunk field specified for the port, inherits openStackMachine.Spec.Trunk.
+		if pOpts.Trunk == nil {
+			pOpts.Trunk = &openStackMachine.Spec.Trunk
+		}
```

Comment:
You have to create the variable `pOpts` to not override `port.Trunk`, correct?
Or can we do sth like this? And pass `PortOpts: &port` in l.164 and l.172

```suggestion
		// No Trunk field specified for the port, inherits openStackMachine.Spec.Trunk.
		if port.Trunk == nil {
			port.Trunk = &openStackMachine.Spec.Trunk
		}
```

Reply:
pOpts is created to avoid code duplication. 
The check needs to be done for each Port over the loop.

Reply:
But you don't have to set `pOpts := &openStackMachine.Spec.Ports[i]` if `port` is already `openStackMachine.Spec.Ports[i]` because of the for loop